src/hotspot/share/memory/heapShared.cpp

```diff
@@ -525,10 +650,91 @@ void HeapShared::serialize_subgraph_info_table_header(SerializeClosure* soc) {
   _run_time_subgraph_info_table.serialize_header(soc);
 }
 
+static void verify_the_heap(Klass* k, const char* which) {
+  if (VerifyArchivedFields) {
+    ResourceMark rm;
+    log_info(cds, heap)("Verify heap %s initializing static field(s) in %s",
+                        which, k->external_name());
+    VM_Verify verify_op;
+    VMThread::execute(&verify_op);
+    if (!FLAG_IS_DEFAULT(VerifyArchivedFields)) {
+      // If this -XX:+VerifyArchivedFields is specified on the command-line, do extra
+      // checks.
```

Comment:
s/this//

Reply:
I think we can keep the code and change the comments to the following:

```
    // If VerifyArchivedFields has a non-default value (e.g., specified on the command-line), do
    // more expensive checks.
```